lib/puppet_x/cisco/cmnutils.rb

```diff
@@ -125,6 +125,38 @@ def self.range_summarize(range_str, sort=true)
       end
     end # class Utils
 
+    # PuppetX::Cisco::PvlanUtils - Common BGP methods used by BGP Types/Providers
+    class PvlanUtils
+      # This api is used by private vlan to prepare the massage the input.
+      # The input can be in the following formats for vlans:
+      # 10-12,14. Prepare_list api is transforming this input into a flat array.
+      # In the example above the returned array will be 10, 11, 12, 13. Prepare
+      # list is first splitting the input on ',' and the than expanding the vlan
+      # range element like 10-12 into a flat array. The final result will
+      # be a  flat array.
+
+      def self.prepare_list(input)
```

Comment:
I removed all but one since for vlan association at vlan level, the return output is  a flat array so the 'should' has to be massage in order to match the resource in the manifest.